File: frontend/src/api/person.ts
import request from './request'
import type { Person } from '@/types'

export interface CreatePersonRequest {
  groupId: string
  firstName: string
  lastName?: string
  gender?: 'MALE' | 'FEMALE' | 'UNKNOWN'
  birthDate?: string
  deathDate?: string
  birthPlace?: string
  currentSpouseId?: string
}

export interface UpdatePersonRequest {
  firstName?: string
  lastName?: string
  gender?: 'MALE' | 'FEMALE' | 'UNKNOWN'
  birthDate?: string
  deathDate?: string
  birthPlace?: string
  currentSpouseId?: string
}

export const personApi = {
  getGroupPersons: (groupId: string): Promise<Person[]> => {
    return request.get(`/persons/group/${groupId}`)
  },
  
  getPerson: (personId: string): Promise<Person> => {
    return request.get(`/persons/${personId}`)
  },
  
  createPerson: (data: CreatePersonRequest): Promise<Person> => {
    return request.post('/persons', data)
  },
  
  updatePerson: (personId: string, data: UpdatePersonRequest): Promise<Person> => {
    return request.put(`/persons/${personId}`, data)
  },
  
  deletePerson: (personId: string): Promise<void> => {
    return request.delete(`/persons/${personId}`)
  },
  
  searchPersons: (groupId: string, keyword: string): Promise<Person[]> => {
    return request.get(`/persons/group/${groupId}/search`, { params: { keyword } })
  }
}

File: backend/src/main/java/com/familytree/domain/Person.java
package com.familytree.domain;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "persons")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Person {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "group_id", nullable = false)
    private UUID groupId;

    @Column(name = "first_name", nullable = false, length = 50)
    private String firstName;

    @Column(name = "last_name", length = 50)
    private String lastName;

    @Enumerated(EnumType.STRING)
    @Column(name = "gender", length = 10)
    private Gender gender;

    @Column(name = "birth_date")
    private LocalDate birthDate;

    @Column(name = "death_date")
    private LocalDate deathDate;

    @Column(name = "birth_place", length = 100)
    private String birthPlace;

    @Column(name = "current_spouse_id")
    private UUID currentSpouseId;

    @Column(name = "created_by")
    private UUID createdBy;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Version
    @Column(name = "version")
    private Integer version;

    @OneToMany(mappedBy = "personId", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Photo> photos;

    @OneToMany(mappedBy = "personId", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<CustomAttribute> customAttributes;

    public enum Gender {
        MALE, FEMALE, UNKNOWN
    }

    public String getFullName() {
        if (lastName != null && !lastName.isEmpty()) {
            return lastName + firstName;
        }
        return firstName;
    }
}
